src/main/java/com/openshift/jenkins/plugins/pipeline/model/GlobalConfig.java

```diff
@@ -1,71 +1,57 @@
 package com.openshift.jenkins.plugins.pipeline.model;
 
 
-// this class is storage for Global Jenkins config, accessible via config.jelly and extensions 
+import com.openshift.jenkins.plugins.pipeline.*;
+
+// this class is storage for Global Jenkins config, accessible via config.jelly and extensions
 // to DescriptorImpl extensions to BuildStepDescriptor<Builder> which we only define in the 
 // freestyle version of our build steps.  They get executed even if no freestyle jobs are defined,
 // so no need to define them twice.  That said, we store the values in this singleton to allow for easy
 // access for the DSL versions of our build steps.
 public class GlobalConfig {
 
-	private static final long MINUTE = 60*1000;
-
-	public static final long UBER_DEFAULT_WAIT = 5*MINUTE;
+    private static final long MINUTE = 60 * 1000;
 
-	public static final long DEFAULT_BUILD_WAIT = 15*MINUTE;
-	public static final long DEFAULT_BUILD_VERIFY_WAIT = 1*MINUTE;
+    public static final long UBER_DEFAULT_WAIT = 5 * MINUTE;
 
-	public static final long DEFAULT_DEPLOY_WAIT = 10*MINUTE;
-	public static final long DEFAULT_DEPLOY_VERIFY_WAIT = 3*MINUTE;
+    public static final long DEFAULT_BUILD_WAIT = 15 * MINUTE;
+    public static final long DEFAULT_BUILD_VERIFY_WAIT = 1 * MINUTE;
 
-	public static final long DEFAULT_SCALER_WAIT = 3*MINUTE;
-	public static final int DEFAULT_SERVICE_VERIFY_RETRY = 100;
+    public static final long DEFAULT_DEPLOY_WAIT = 10 * MINUTE;
+    public static final long DEFAULT_DEPLOY_VERIFY_WAIT = 3 * MINUTE;
 
-	public static final long DEFAULT_EXEC_WAIT = 3*MINUTE;
+    public static final long DEFAULT_SCALER_WAIT = 3 * MINUTE;
+    public static final int DEFAULT_SERVICE_VERIFY_RETRY = 100;
 
-	private static long buildWait = DEFAULT_BUILD_WAIT;
-	private static long buildVerifyWait = DEFAULT_BUILD_VERIFY_WAIT;
-	private static long deployWait = DEFAULT_DEPLOY_WAIT;
-	private static long deployVerifyWait = DEFAULT_DEPLOY_VERIFY_WAIT;
-	private static long scalerWait = DEFAULT_SCALER_WAIT;
-	private static long execWait = DEFAULT_EXEC_WAIT;
+    public static final long DEFAULT_EXEC_WAIT = 3 * MINUTE;
 
-	private static int serviceVerifyRetry = DEFAULT_SERVICE_VERIFY_RETRY;
+    public static long getBuildWait() {
+        return new OpenShiftBuilder.DescriptorImpl().getConfiguredDefaultWaitTime();
```

Comment:
A comment somewhere ... maybe the abstract descriptor class you created, would be good here, as I think we've fallen into the realm of Jenkins black magic again.  

If I'm reading the tea leaves correctly, but instantiating the DescriptorImpl, you are forcing a load of the associated global config that was saved previously to occur.  Is that it?

(Is so, very clever diagnosis / solution hacking :-) ... was this solution even googable ?)